app/src/ui/create-branch/create-branch-dialog.tsx

```diff
@@ -412,6 +412,12 @@ export class CreateBranch extends React.Component<
       this.props.repository
     )
 
+    // Make sure user branch name hasn't changed during parsing of repo rules
+    // (async due to a config retrieval of users with commit signing repo rules)
+    if (this.state.branchName !== branchName) {
```

Comment:
From cherry-picked merge commit from https://github.com/desktop/desktop/pull/17394 to get the last commit that wasn't shipped to beta on Friday. (Minor change)